### ISSUE TITLE
Option to choose an input device

### ISSUE THREAD

**Issue by SurpriseSnek** [milkdrop2077/MilkDrop3#10]
This project is awesome however I was wondering if there was a way to use it with a different audio source as it doesn't work very well with my audio setup. That's because I use voicemeter banana to mix several inputs together and out to my interface. Spotify is one of the inputs I've separated out, so it's not playing on a default device, but instead over a virtual cable. Being able to choose an input would make this pretty tidy, but right now it's a bit annoying to work with. If there is already a way to just set this then my bad, and thanks for any help.

**Comment by milkdrop2077** [comment]
I'll look into the ability to choose a different audio source soon.
Can you see the source you want to use with the command line: loopback-capture --list-devices
https://matthewvaneerde.wordpress.com/2008/12/16/sample-wasapi-loopback-capture-record-what-you-hear/ ?

**Comment by SurpriseSnek** [comment]
Apologies for the delay, life got in the way. I am able to see the device using that program. If you need me to post the list or something I can do that.

**Comment by milkdrop2077** [comment]
Can you see the device that you want to use?

**Comment by SurpriseSnek** [comment]
Yes, I'm able to see the device I want to use, so hopefully it should be accessible.

**Comment by milkdrop2077** [comment]
ok that's good news if you can see the device then I should be able to let you choose it. I'll add a way to choose the audio device this summer hopefully

**Comment by OfficialIncubo** [comment]
Really excited about this feature! I press Alt+A to change the audio input! I wish I had this feature on my BeatDrop (and Milkdrop3)!

**Comment by Energyblast** [comment]
Looking forward to this as well. I can only use it if I can select audio source, I have a lot going on in the background.

**Comment by milkdrop2077** [comment]
So I worked on that, and I found 2 problems:
- microphone input is not supported by beatdrop, so milkdrop3.
- high res audio like 192kHz is not supported, and 96kHz is partially supported  (thanks to William R. for the tests)

I think it will simply be better to port the much more advanced audio capture part from Projectm, but that will take time

**Comment by OfficialIncubo** [comment]
> microphone input is not supported by beatdrop, so milkdrop3.

Milkdrop3 is based of BeatDrop, but why did you say that? 🙄

Did you try listing all the audio devices using cmd and loopback-capture?
Does loopback-capture lists microphone inputs?

**Comment by milkdrop2077** [comment]
I've try forcing MilkDrop3 to react to the microphone but it's not working. I'm not sure what the separate streams is..

**Comment by OfficialIncubo** [comment]
I need this feature to be done on my BeatDrop impromevent. I'm so excited.

**Comment by kabouterPop** [comment]
Just wanted to add support for this. I was hoping to run MilkDrop while recording audio in my DAW (Reaper) but the signal doesn't reach MD, even though my Focusrite i818 shows the sound reaching the PC. Even with Loopback on I can't get the audio to reach MD. It's probably because I'm using ASIO. I also thought it could be cool to have a MilkDrop plug-in for the DAW using one of the newer (non-proprietary) plugin formats, like CLAP to make a guess.

**Comment by oPTiK-aL** [comment]
First off major props @milkdrop2077 for an awesome project and easily a top 5 visualization application (no longer just a plug-in). Keep up the amazing work!!

Now onto my question --> I did want to check in on any updates with MIC input support and if there has any progress been made on it. I listen to a lot of vinyl and it would be really cool to have Milkdrop visualizations on my monitor / TV synced up to the music playing off the turntable. If not natively supported by Milkdrop, are there any workarounds I could try?
